### PR TITLE
Add copy saved objects among workspaces functionality

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
The main change of this PR is adding the copy saved objects among workspaces functionality, the UI is as follows:

Copy button:
image

Copy dialog:
image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+              defaultMessage="Cancel"
+            />
+          </EuiButtonEmpty>
+
+          <EuiButton
+            fill
+            data-test-subj="copyConfirmButton"
+            onClick={this.copySavedObjects}
+            isLoading={this.state.isLoading}
+            disabled={!confirmCopyButtonEnabled}
```
[the remark this answers is not in the document]
Comment (gaobinlong) [reply src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx:309]: Yeah, I'll change it.

File: src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx [hunk 2]
```diff
@@ -106,6 +109,7 @@ export interface SavedObjectsTableProps {
   savedObjectsClient: SavedObjectsClientContract;
   indexPatterns: IndexPatternsContract;
   http: HttpStart;
+  workspacesStart: WorkspacesStart;
```
[the remark this answers is not in the document]
Comment (gaobinlong) [reply src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx:112]: That's better, I'll change it.

File: src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+  }
+
+  workspaceToOption = (workspace: WorkspaceAttribute): WorkspaceOption => {
+    return { label: workspace.name, key: workspace.id, value: workspace };
+  };
+
+  async componentDidMount() {
+    const { workspacesStart } = this.props;
+    const workspaceList = await workspacesStart.client.workspaceList$;
+    const currentWorkspace = await workspacesStart.client.currentWorkspace$;
```
[the remark this answers is not in the document]
Comment (gaobinlong) [reply src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx:82]: I think the copy operation does not take too much time to execute, and users will not stay in the dialog too much time, so maybe it's no need to refresh the dropdown list when the workspace list changes.

File: src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaobinlong) [reply src/plugins/saved_objects_management/public/management_section/objects_table/components/copy_modal.tsx:309]: By my test, found that when isLoading is true, the button is also disabled, like this:
image

**Review by wanglam — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**